### PR TITLE
Feature/cleanup

### PR DESCRIPTION
**Pull request by malaffoon**
The goal is to make this project more consistent with the other RDW projects wrt remote config, docker, unit tests, coverage, etc.

I didn't have much success wrt security and running with local, non-SAML auth. And i couldn't get the JKS file served up from the configuration server properly. Oh, and because of that, i haven't pushed forward on dockerization yet. But some minor tweaks ...

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+## RDW_Reporting for Developers
```
Comment (malaffoon) [inline CONTRIBUTING.md]: Trying to separate true developer issues from non-developer consumers of this.

File: src/main/java/org/opentestsystem/rdw/reporting/security/User.java [hunk 2]
```diff
@@ -48,7 +49,7 @@ public boolean isAccountNonLocked() {
 
     @Override
     public boolean isCredentialsNonExpired() {
-        return false;
+        return credentialsNonExpired;
```
Comment (malaffoon) [inline src/main/java/org/opentestsystem/rdw/reporting/security/User.java]: actually found a bug in all of this

File: src/main/java/org/opentestsystem/rdw/reporting/security/WebSecurityConfig.java [hunk 3]
```diff
@@ -1,11 +1,5 @@
-package org.opentestsystem.rdw.reporting.config;
```
Comment (malaffoon) [inline src/main/java/org/opentestsystem/rdw/reporting/security/WebSecurityConfig.java]: oops, most of this is unintentional cmd-alt-O reformatting.

File: src/main/webapp/.gitignore [hunk 4]
```diff
@@ -5,9 +5,9 @@
 /tmp
 
 # dependencies
-/node_modules
```
Comment (malaffoon) [inline src/main/webapp/.gitignore]: this wasn't working on my system ... i was getting 16000+ unversioned files.

File: build.gradle [hunk 5]
```diff
@@ -85,9 +92,9 @@ bootRun {
         jvmArgs project.jvmArgs.split('\\s+')
     }
 }
-bootRepackage {
-    mainClass = 'org.opentestsystem.rdw.reporting.Application'
-}
+
+// generate coverage report automatically (./build/reports/jacoco/test/html/index.html)
```
Comment (agorina) [inline build.gradle]: Yeah!!!!

File: src/main/resources/application.yml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+spring:
+  datasource:
+    url: jdbc:mysql://localhost:3306/reporting?useSSL=false
+    username: root
+    password:
+    driverClassName: com.mysql.jdbc.Driver
+    testWhileIdle: true
+    validationQuery: SELECT 1
+    initialize: false
```
Comment (agorina) [inline src/main/resources/application.yml]: Since you introduced this one, the yml in the test folder can be deleted.

Comment (malaffoon) [inline src/main/resources/application.yml]: good point, will do

**Pull request merged**